test/org/zaproxy/zap/extension/pscanrulesAlpha/InsecureFormLoadScannerUnitTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class InsecureFormLoadScannerUnitTest extends PassiveScannerTest {
+
+	private HttpMessage msg;
+
+	@Before
+	public void before() throws URIException {
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		requestHeader.setURI(new URI("http://example.com", false));
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+		msg.getResponseHeader().setStatusCode(HttpStatusCode.OK);
+		msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/html");
+	}
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new InsecureFormLoadScanner();
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseIsNotHttp() throws URIException {

```

Comment:
`IsHttps` (and `setSecure(true)`)?

Reply:
I'll have to go through it again. You're probably right, I admit keeping FormPost and FormLoad scanners straight was causing me issues at the time :grinning: